### PR TITLE
[various] Update minimum iOS versions to 12.0

### PR DESCRIPTION
**Pull request by stuartmorgan**
This makes the following mass changes:
- Builds all examples with current stable (3.16.7), to pick up the auto-migration of projects to a minimum of iOS 12.
  - In some cases this picks up other boilerplate Xcode project migrations that had never been checked in to those projects.
- Updates all iOS plugin implementations to indicate that they require iOS 12.0 or later.
- Updates all the READMEs of the app-facing versions of those plugins to indicate that iOS 11 will no longer receive implementation updates.

The only code changes are:
- the removal of some `@available` checks in `in_app_purchase_storekit` that had become no-ops with the support change
- the removal of a `!` in `camera_avfoundation` Dart code, which is no longer necessary due to changes in nullable field promotion in newer versions of Dart.

These changes are version-exempt [per repo policy](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates), so the changelog changes all use NEXT.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by gspencergoog — APPROVED** [review]

File: packages/google_maps_flutter/google_maps_flutter/example/ios/Runner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -397,10 +397,12 @@
 			inputPaths = (
 				"${PODS_ROOT}/Target Support Files/Pods-Runner/Pods-Runner-resources.sh",
 				"${PODS_ROOT}/GoogleMaps/Maps/Frameworks/GoogleMaps.framework/Resources/GoogleMaps.bundle",
+				"${PODS_CONFIGURATION_BUILD_DIR}/google_maps_flutter_ios/google_maps_flutter_ios_privacy.bundle",
```
Comment (jmagman) [inline packages/google_maps_flutter/google_maps_flutter/example/ios/Runner.xcodeproj/project.pbxproj:400]: I didn't realize adding the privacy bundle would update the host Xcode project, not just the generated Pod project.

**Comment by stuartmorgan** [comment]
Changelog override: the packages missing changelog changes just have uninteresting example project changes.

**Review by jmagman — APPROVED** [review]
This change LGTM, thanks for doing it.

**Review by Piinks — APPROVED** [review]
LGTM!

**Pull request merged**